agent/stats/service_connect_linux.go

```diff
@@ -0,0 +1,90 @@
+//go:build linux
+
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package stats
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+	"sync"
+
+	"github.com/cihub/seelog"
+
+	apitask "github.com/aws/amazon-ecs-agent/agent/api/task"
+)
+
+const (
+	statsPort         = ":9901/stats/prometheus"
+	bridgeNetworkMode = "bridge"
+)
+
+type ServiceConnectStats struct {
+	//TODO Change the type of Service Connect stats when it is defined.
+	stats string
+	lock  sync.RWMutex
+}
+
+func newServiceConnectStats() (*ServiceConnectStats, error) {
+	return &ServiceConnectStats{}, nil
+}
+
+func (sc *ServiceConnectStats) retrieveServiceConnectStats(task *apitask.Task, wg *sync.WaitGroup) {
+	defer wg.Done()
+	var ipAddress string
+	if task.IsNetworkModeAWSVPC() {
+		ipAddress = task.GetLocalIPAddress()
+	} else {
+		for _, container := range task.Containers {
+			// TODO: Implement a proper check for appnet agent container when it is defined
```

Comment:
can we add a name convention for SC TODOS, I was thinking `// TODO [SC]: ...`

Reply:
makes sense.